src/sensors/sensor.h

```diff
@@ -98,6 +101,16 @@ class Sensor {
 	bool hasNewDataToSend() { return newFusedRotation || newAcceleration; };
 	inline bool hasCompletedRestCalibration() { return restCalibrationComplete; }
 
+	virtual uint8_t getDataType() { return SENSOR_DATATYPE_ROTATION; };
+
+	uint16_t getSensorPosition() { return m_SensorPosition; };
+
+	void setSensorInfo(uint16_t sensorPosition) { m_SensorPosition = sensorPosition; };
+
+	TPSCounter m_tpsCounter;
+	TPSCounter m_dataCounter;
+	std::shared_ptr<SlimeVR::SensorInterface> m_hwInterface;
```

Comment:
this probably should be a unique_ptr, at least I don't see it being copied anywhere